crescent/event.py

```diff
@@ -46,7 +48,7 @@ def event(
     if not iscoroutinefunction(callback):
         raise ValueError(f"`{callback.__name__}` must be an async function.")
 
-    def hook(self: MetaStruct[CallbackT, None]):
+    def hook(self: MetaStruct[CallbackT[Any], None]) -> None:
         self.app.subscribe(event_type=unwrap(event_type), callback=self.callback)
 
-    return MetaStruct(callback=callback, metadata=None, app_set_hooks=[hook])
+    return MetaStruct(callback=callback, metadata=None, app_set_hooks=[hook])  # type: ignore
```

Comment:
what does the `# type: ignore` do

Reply:
Because `callback`'s type is incorrect. I'll change this to a cast to be more clear (and add a comment).